frontend/src/features/contacts/contact/detail/Organization.tsx

```diff
@@ -0,0 +1,215 @@
+import { FormSection } from 'components/common/form/styles';
+import { ContactMethodTypes } from 'constants/contactMethodType';
+import { Dictionary } from 'interfaces/Dictionary';
+import {
+  IContactAddress,
+  IContactMethod,
+  IContactOrganization,
+  IContactPerson,
+} from 'interfaces/IContact';
+import * as React from 'react';
+import { Col } from 'react-bootstrap';
+import { FaCircle, FaRegBuilding } from 'react-icons/fa';
+import { Link } from 'react-router-dom';
+import { phoneFormatter } from 'utils/formUtils';
+
+import * as Styled from '../../styles';
+
+export interface OrganizationViewProps {
+  organization: IContactOrganization;
+}
+
+interface ContactInfoField {
+  info: string;
+  label: string;
+}
+
+interface AddressField {
+  address: IContactAddress;
+  label: string;
+}
+
+const OrganizationView: React.FunctionComponent<OrganizationViewProps> = ({ organization }) => {
+  const phoneTypes: Dictionary<string> = {};
+  phoneTypes[ContactMethodTypes.WorkMobil] = 'Mobile';
+  phoneTypes[ContactMethodTypes.WorkPhone] = 'Work';
+  phoneTypes[ContactMethodTypes.PersPhone] = 'Home';
+  phoneTypes[ContactMethodTypes.Fax] = 'Fax';
+
+  const personPhoneNumbers: ContactInfoField[] = getContactInfo(organization, phoneTypes);
+
+  const emailTypes: Dictionary<string> = {};
+  emailTypes[ContactMethodTypes.WorkEmail] = 'Work';
+  emailTypes[ContactMethodTypes.PerseEmail] = 'Personal';
+  const personEmails: ContactInfoField[] = getContactInfo(organization, emailTypes);
+
+  let personAddresses: AddressField[];
+  if (organization.addresses === undefined) {
+    personAddresses = [];
+  } else {
+    personAddresses = organization.addresses.reduce(
+      (accumulator: AddressField[], value: IContactAddress) => {
+        accumulator.push({ label: value.addressType.description, address: value });
+
+        return accumulator;
+      },
+      [],
+    );
+  }
+
+  return (
+    <>
+      <FormSection key={'contact-org-' + organization.id + '-names'} className="mb-4">
+        <Styled.RowAligned>
+          <Col>
+            <Styled.H2>
+              <FaRegBuilding size={20} className="mr-2" />
+              <span data-testid="contact-organization-fullname">{organization.name}</span>
```

Comment:
try to avoid data-testid where possible. react testing library's philosophy is:
**The more your tests resemble the way your software is used, the more confidence they can give you.**
data-testid's are not how your user navigates the application, so avoid where possible

Reply:
Interesting, I followed the example on the react site about testing, and most of the examples were using testing ids.
In regards to how is navigated, I would not try to simulate the user navigation or interaction with the page at the unit test level. In my experience front end unit-tests tent to become quite convoluted since the user navigates the page visually while the testing framework is doing it as markup. We should chat more about this to see what provides the best value both reading the tests and how maintainable they are